### DISCUSSION TITLE
Firefox strict and custom fonts

### DISCUSSION THREAD

**Discussion by slvrbuu** [yokoffing/Betterfox#261] [Q&A] [answered]
Hello, I've just started using this and I noticed it forces Firefox into strict, however under strict it doesn't allow my custom font that I use to help better read heavy font pages. I have dyslexia and I kind of need it to make my experience easier. This seems to be a result of enabling 'suspected fingerprinters'. So I'm wondering if there is a way to bypass this as an override setting, or if I'm just forced to sit in custom and use everything but suspected fingerprinters. Thanks!

**Comment by yokoffing** [comment, marked as answer]
Mozilla recently replaced the font visibility prefs and increased fingerprinting protections. Your way works, but **Custom** omits other privacy protections.

### Option 1
You can disable font protection this way:

1. Change ETP back to **Strict**
2. Go to `about:config`
3. Edit `privacy.fingerprintingProtection.overrides` to `-FontVisibilityLangPack`

### Option 2
If that by itself doesn't work, repeat the steps, but this time edit the prefence to `-FontVisibilityBaseSystem,-FontVisibilityLangPack`.

@slvrbuu Please let me know if Option 1 was sufficient, or if you needed Option 2.

**Reply by slvrbuu** [reply]
Hey there, thank you very much for your response, and suggestion!

The first method does seem to have done the trick. I have strict on, and my font is as I want it. I very much appreciate the help.

**Reply by dcog989** [reply]
This fix works for me initially, but Firefox is doing something in the background so that it soon stops *some* installed fonts from displaying. E.g. Arial is not displayed for me even though it is installed on my Windows machine. Firefox clean profile, Edge, and Chromium all display Arial.

Any other settings to tweak?

**Reply by Shoosh08** [reply]
> This fix works for me initially, but Firefox is doing something in the background so that it soon stops _some_ installed fonts from displaying. E.g. Arial is not displayed for me even though it is installed on my Windows machine. Firefox clean profile, Edge, and Chromium all display Arial.
> 
> Any other settings to tweak?

After weeks of pain, I managed to find the solution via Firefox's article. You don't need to play with "privacy.fingerprintingProtection.overrides" in config, just keep it default.
Go to Settings -> Privacy & Security -> Custom -> Set "Suspected fingerprinters" to "Only in private windows".

**Reply by yokoffing** [reply]
@Shoosh08 Going the **Custom** route doesn't activate other **Strict** protections (e.g., go to Securefox and `CTRL`+`F` the phrase `enabled with "Strict"` to see the protections **Strict ETP** activates), but you may be okay with that.

You can also omit FPP from **Strict ETP** by following the steps I added here recently, at least in theory. I need more testing of the opt out mechanism: https://github.com/yokoffing/Betterfox/wiki/Common-Overrides#standard-tracking-protection

FPP has certainly evolved since I last posted about this in late 2023. There are ways to be even more modular by altering FPP itself, hence AF's description here: `https://github.com/arkenfox/user.js/blob/c25187332e19c633b956a6755a7971a6591d3e8d/user.js#L741-L756` However, he has not listed — and I haven't found — the value that obfuscates fonts.

**Reply by Shoosh08** [reply]
> @Shoosh08 Going the **Custom** route doesn't activate other **Strict** protections (e.g., go to Securefox and `CTRL`+`F` the phrase `enabled with "Strict"` to see the protections **Strict ETP** activates), but you may be okay with that.
> 
> You can also omit FPP from **Strict ETP** by following the steps I added here recently, at least in theory. I need more testing of the opt out mechanism: https://github.com/yokoffing/Betterfox/wiki/Common-Overrides#standard-tracking-protection
> 
> FPP has certainly evolved since I last posted about this in late 2023. There are ways to be even more modular by altering FPP itself, hence AF's description here: `https://github.com/arkenfox/user.js/blob/c25187332e19c633b956a6755a7971a6591d3e8d/user.js#L741-L756` However, he has not listed — and I haven't found — the value that obfuscates fonts.

Thank you very much for your time and support! Much appreciated. I did according to your reply from Nov 2023 but sadly my custom fonts are not taking effect.
I took a look at all of the settings you've mention in the Securefox with the phrase `enabled with "Strict"` and they were a total of nine. Only one of them was turned off for me with my Custom setting and that is "privacy.fingerprintingProtection". I don't mind losing that protection for much better-looking fonts.
Best regards!

**Reply by yokoffing** [reply]
@Shoosh08 I'm glad that works for you; I totally understand. One of the best thing about Firefox is how modular it is — you can customize it to your liking. Using [uBlock Origin](https://addons.mozilla.org/blog/ublock-origin-everything-you-need-to-know-about-the-ad-blocker/) and maybe some [Privacy-focused lists](https://github.com/yokoffing/filterlists?tab=readme-ov-file#privacy) should take care of most third-party trackers that are fingerprinting anyway.

Understanding more on how to alter Strict ETP and FPP is on my 'to do' list.
